R/api-bump-version.R

```diff
@@ -5,6 +5,28 @@
 #'
 #' @inheritParams update_version
 #' @export
+#'
+#' @examples
+#' # Create temporary directory to hold a mock package.
+#' tempdir <- file.path(tempdir(check = TRUE), "fledge")
+#' dir.create(tempdir)
+#' # Create said mock package.
+#' # Set open to TRUE if you want to play in the mock package.
+#' pkg <- create_fledge_mock_repo(open = FALSE, news = TRUE, tempdir = tempdir)
+#' # Use functions as if inside the newly created package project.
+#' # (Or go and actually run code inside the newly created package project!)
+#' usethis::with_project(pkg, {
```

Comment:
Should we manually record the current project and use `proj_activate()` ? This might be simpler to understand, and allows stepping through the code line by line.

Reply:
So, I started doing this and the code ends up being much more cumbersome because one needs to activate the project _and_ change the directory / use the repo argument for gert functions (gert does not use the active project, it uses the current directory by default). The latter (using the repo argument for gert functions) is not a good idea as it means one can't run the exact same code from within the original project vs in the mock project (I assume some users might try out the example code there).

Besides, the code to record and reset the current project would need to take into account the fact that the user is not necessarily in any project.

I can still do it, but wanted to explain the caveats before.

```r
# Create temporary directory to hold a mock package.
tempdir <- file.path(tempdir(check = TRUE), "fledge")
dir.create(tempdir)
# Create said mock package.
# Set open to TRUE if you want to play in the mock package.
pkg <- create_demo_project(open = FALSE, news = TRUE, dir = tempdir)
# Use functions as if inside the newly created package project.
# (Or go and actually run code inside the newly created package project!
current_proj <- usethis::proj_get()
current_dir <- getwd()
usethis::proj_set(pkg)
setwd(pkg)
# Add a new R file.
usethis::use_r("cool-function", open = FALSE)
# Pretend we added useful code inside it.
# Track the new R file with Git.
gert::git_add("R/cool-function.R")
gert::git_commit("- Add cool function.")
# Bump version with fledge.
bump_version()
# Recreate original state
try(usethis::proj_set(current_proj), silent = TRUE)
setwd(current_dir)
# Clean the temporary directory we had created
unlink("tempdir", recursive = TRUE)
```


I tend to think that the best and clearer examples will be the ones living in vignettes. :sweat_smile:

Reply:
I see, let's keep the "with" approach then. How about implementing `with_example_project()` just for the purpose of examples, and pointing to e.g. `vignette("fledge")` for a more detailed walkthrough?

Reply:
What would `with_example_project()` do?

Reply:
Combine `create_demo_project()` and `with_project()` . I'm not sure it's a good idea though.

Reply:
I'll open an issue to track this, but not tackle this in this PR (that's already quite big :sweat_smile: )